### PR TITLE
feat: add parameter for custom headers to createJWT

### LINKED ISSUE #65: `JWTPayload` is filtering out custom JWT properties
**Describe the bug**

`decodeJWT` returns a `JWTPayload` which defines a strict set of properties it may contain.
When a JWT with custom properties is being decoded then the decoded payload will not see these properties.

**To Reproduce**

use `did-jwt` in a typescript project, like so:
```typescript
import { decodeJWT } from 'did-jwt'

//a JWT with a `status` property in the payload
const decoded = decodeJWT('eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NkstUiJ9.eyJpYXQiOjE1NzIzNTE2NTYsInN0YXR1cyI6eyJpZCI6InJpbmtlYnk6MHhTdGF0dXNSZWdpc3RyeUFkZHJlc3MiLCJ0eXBlIjoiRXRoclN0YXR1c1JlZ2lzdHJ5MjAxOSJ9LCJpc3MiOiJkaWQ6ZXRocjoweGYzYmVhYzMwYzQ5OGQ5ZTI2ODY1ZjM0ZmNhYTU3ZGJiOTM1YjBkNzQifQ.OoR164SxkZEl0Q0aW7yBkKpAnWL4DYeiBtN85L4tD793mfhxb_Q2rxPot0DUFXVnBULoS55x5MYuRMVRpOSl7QE')

console.log(decoded.payload.status)
// compilation fails with` Property 'status' does not exist on type 'JWTPayload'.`
// a workaround is to cast to `any`
```

**Expected behavior**

`JWTPayload` and `JWTHeader` should allow custom properties.
Compilation should not fail in the scenario above

**Environment**

- did-jwt version 3.0.0

### PR DESCRIPTION
**Pull request by nikolockenvitz**
also fixes #65

I putted everything into one commit as discussed in #73.
I am not sure whether there is further documentation which should be updated or comments that should be adjusted.

Thanks you very much for your help, @mirceanis

### CODE REVIEW

File: docs/guides/index.md [hunk 1]
```diff
@@ -27,20 +27,21 @@ createJWT(
...
+createJWT(payload, options, header)
 ```
 
 Name | Description | Required
 ---- | ----------- | --------
 `payload` | an object containing any claims you want to encode in the JWT including optional standard claims such as `sub`, `aud` and `exp` | yes
-`settings.issuer` | The [DID](https://w3c-ccg.github.io/did-spec/#decentralized-identifiers-dids) of the audience of the JWT | yes
-`settings.signer` | A signing function (see SimpleSigner) | yes
-`settings.expiresIn` | How many seconds after signing should the JWT be valid (sets the `exp` claim) | no
+`options.issuer` | The [DID](https://w3c-ccg.github.io/did-spec/#decentralized-identifiers-dids) of the audience of the JWT | yes
```
Comment (mirceanis) [inline docs/guides/index.md:36]: It seems there was a mistake in the original doc here
```suggestion
`options.issuer` | The [DID](https://w3c-ccg.github.io/did-spec/#decentralized-identifiers-dids) of the issuer of the JWT | yes
```

File: src/JWT.ts [hunk 2]
```diff
@@ -153,16 +155,19 @@ export function decodeJWT(jwt: string): JWTDecoded {
  *  @param    {String}            options.issuer      The DID of the issuer (signer) of JWT
  *  @param    {String}            options.alg         The JWT signing algorithm to use. Supports: [ES256K, ES256K-R, Ed25519], Defaults to: ES256K
  *  @param    {SimpleSigner}      options.signer      a signer, reference our SimpleSigner.js
+ *  @param    {Object}            header             optional object to inject custom headers
```
Comment (mirceanis) [inline src/JWT.ts:158]:
```suggestion
 *  @param    {Object}            header             optional object to specify or customize the JWT header
```

File: src/JWT.ts [hunk 3]
```diff
@@ -153,16 +155,19 @@ export function decodeJWT(jwt: string): JWTDecoded {
  *  @param    {String}            options.issuer      The DID of the issuer (signer) of JWT
  *  @param    {String}            options.alg         The JWT signing algorithm to use. Supports: [ES256K, ES256K-R, Ed25519], Defaults to: ES256K
```
Comment (mirceanis) [inline src/JWT.ts:156]:
```suggestion
 *  @param    {String}            options.alg         [DEPRECATED] The JWT signing algorithm to use. Supports: [ES256K, ES256K-R, Ed25519], Defaults to: ES256K.
 *                                                    Please use `header.alg` to specify the algorithm
```

**Comment by nikolockenvitz** [comment]
Perfect, thank you very much for your help @mirceanis

**Pull request merged**